thunder/__init__.py

```diff
@@ -442,6 +442,11 @@ def get_computation_and_inputs(*args, **kwargs):
         # which seems to break the consistency of cache_info, leading to a failure in cache_info check.
         cache_info["alias_tensor_indices"] = _alias_tensor_of_args_kwargs(*args, **kwargs)
 
+        # Store the `is_grad_enabled` state of PyTorch. This is used by vjp transform
+        # to treat certain Symbols as constant.
+        cache_info["is_grad_enabled"] = pytorch.is_grad_enabled()
```

Comment:
Is this cache_info entry needed?

Reply:
Well if you call with grad enabled and then without, you would want to have a cache miss?

Reply:
If we think we don't need it, let's remove it in a follow-up.

Reply:
It is for the following cases

Example 1
```python
jfn = thunder.jit(fn)
with torch.no_grad():
    jfn(x)   # This will be compiled with no_grad

jfn(x)   # We want this to be recompiled.
```

Example 2
```python
jfn = thunder.jit(fn)

jfn(x)

with torch.no_grad():
    jfn(x)   # We want this to be recompiled
```

Reply:
> Well if you call with grad enabled and then without, you would want to have a cache miss?

Of course. 

How does this work if the content of `cache_info["is_grad_enabled"]` is not checked anywhere in this pull request?

Reply:
In `jit_ext.py`, we read the values from cache_info and add corresponding checks in prologue.

https://github.com/Lightning-AI/lightning-thunder/blob/11a32a4b9575acc6de670450e1c0d66cab541305/thunder/core/jit_ext.py#L1580-L1598

Reply:
Thank you!